docs(auth): add a section about user invites (#47774)
supabase/supabase: apps/docs/content/guides/auth/users.mdx

```js
import { createClient } from '@supabase/supabase-js'

// Use your project's secret key (sb_secret_...), and only ever on a trusted server.
const supabase = createClient(process.env.SUPABASE_URL, process.env.SUPABASE_SECRET_KEY, {
  auth: {
    autoRefreshToken: false,
    persistSession: false,
    detectSessionInUrl: false,
  },
})

const { data, error } = await supabase.auth.admin.inviteUserByEmail('someone@example.com', {
  data: { name: 'Jane' }, // optional, stored in user_metadata
  redirectTo: 'https://example.com/welcome', // optional, where the invite link sends the user
})
```

<Admonition type="caution">

The secret key (`sb_secret_...`, which replaces the legacy `service_role` key) bypasses Row Level Security and must only be used in a secure server environment. Never expose it in a browser or any publicly accessible client.

</Admonition>

<Admonition type="note">

> The `redirectTo` URL must be in your project's [allowed redirect URLs](/docs/guides/auth/redirect-urls). If it isn't, the invite link falls back to your Site URL instead of the URL you specified.

Reviewer: ```suggestion
The `redirectTo` URL must be in your project's [allowed redirect URLs](/docs/guides/auth/redirect-urls) configuration,. If it isn't, the invite link will fallback to your Site URL instead.
```

Author: Rephrased.

</Admonition>

> The invitation email uses the **Invite user** email template, which you can customize. See [Email Templates](/docs/guides/auth/auth-email-templates) to learn more.

Reviewer 2: ```suggestion
The invitation email uses the **Invite user** email template, which you can customize. Refer to [Email Templates](/docs/guides/auth/auth-email-templates) to learn more.
```

<Admonition type="caution">

> Invitation links expire after the duration set by the **Email OTP Expiration** setting (**Authentication > Sign In / Providers > Auth Providers > Email > Email OTP expiration**), which defaults to 24 hours. This is the same value used for [email OTPs](/docs/guides/auth/auth-email-passwordless#enabling-email-otp), magic links, and other email confirmation links. If an invitation expires before it's accepted, send the user a new invite.

Reviewer: ```suggestion
Invitation links expire by the value set on **Email OTP Expiration** setting (---maybe a link is better here---). the default is 24 hours.

The same value is used for [email OTPs](/docs/guides/auth/auth-email-passwordless#enabling-email-otp), magic links, and other email confirmation links. When an invitation expires before it's accepted, send the user a new invite.
```

Author: Changed to the link.
